nixos/modules/virtualisation/proxmox-image.nix

```
@@ -0,0 +1,168 @@
+
+with lib;
+
+{
+  options = {
+    proxmox = {
+      qemuConf = {
+        # essential configs
+        boot = mkOption {
+          type = types.str;
+          default = "";
+          example = "order=scsi0;net0";
+          description = ''
+            Default boot device. PVE will try all devices in its default order if this value is empty.
+          '';
+        };
+        scsihw = mkOption {
+          type = types.str;
+          default = "virtio-scsi-pci";
+          example = "lsi";
+          description = ''
+            SCSI controller type. Must be one of the supported values given in https://pve.proxmox.com/wiki/Qemu/KVM_Virtual_Machines
+          '';
+        };
+        virtio0 = mkOption {
+          type = types.str;
+          default = "local-lvm:vm-9999-disk-0";
+          example = "ceph:vm-123-disk-0";
+          description = ''
+            Configuration for the default virtio disk. It can be used as a cue for PVE to autodetect the target sotrage.
+            This parameter is required by PVE even if it isn't used.
+          '';
+        };
+        ostype = mkOption {
+          type = types.str;
+          default = "l26";
+          description = ''
+            Guest OS type
+          '';
+        };
+        cores = mkOption {
+          type = types.int;
+          default = 1;
+          description = ''
+            Guest core count
+          '';
+        };
+        memory = mkOption {
+          type = types.int;
+          default = 1024;
+          description = ''
+            Guest memory in MB
+          '';
+        };
+
+        # optional configs
+        name = mkOption {
+          type = types.str;
+          default = "nixos-${config.system.nixos.label}";
+          description = ''
+            VM name
+          '';
+        };
+        net0 = mkOption {
+          type = types.str;
+          default = "virtio=00:00:00:00:00:00,bridge=vmbr0,firewall=1";
+          description = ''
+            Configuration for the default interface. When restoring from VMA, check the
+            "unique" box to ensure device mac is randomized.
+          '';
+        };
+        serial0 = mkOption {
+          type = types.enum [ "socket" ];
+          default = "socket";
+          description = ''
+            Enable serial terminal
+          '';
+        };
+        agent = mkOption {
+          type = types.enum [ "0" "1" ];
+          default = "1";
+          description = ''
+            Expect guest to have qemu agent running
+          '';
+        };
+      };
+      qemuExtraConf = mkOption {
+        type = types.attrsOf types.anything;
+        default = {};
+        example = literalExpression ''{
+          cpu = "host";
+          onboot = 1;
+        }'';
+        description = ''
+          Additional options appended to qemu-server.conf
+        '';
+      };
+      filenameSuffix = mkOption {
+        type = types.str;
+        default = "${config.proxmox.qemuConf.name}";
```

The interpolation in unnecessary here:
```suggestion
        default = config.proxmox.qemuConf.name;
```